test/TestUtils.py

```diff
@@ -185,3 +186,32 @@ def test_expand_paths_vars(monkeypatch):
     monkeypatch.setenv('TEST_PATH', test_path)
     assert utils.expand_paths_vars(['~']) == [os.path.expanduser('~')]
     assert utils.expand_paths_vars(['$TEST_PATH']) == [test_path]
+
+
+def test_get_yaml_files_no_git_verbose(monkeypatch, capsys):
+    options, _ = cli.get_config(['-v'])
+    monkeypatch.setenv('PATH', '')
+    utils.get_yaml_files(options)
+    stdout = capsys.readouterr().out.split('\n')
+    assert stdout[0] == 'Discovering files to lint: git ls-files *.yaml *.yml'
+    assert stdout[1].startswith('Warning: Failed to locate command:')
+
+
+def test_get_yaml_files_not_git_verbose(monkeypatch, capsys):
+    options, _ = cli.get_config(['-v'])
+    monkeypatch.chdir('/tmp')
+    utils.get_yaml_files(options)
+    stdout = capsys.readouterr().out.split('\n')
+    assert stdout[0] == 'Discovering files to lint: git ls-files *.yaml *.yml'
+    assert stdout[1].startswith('Warning: Failed to discover yaml files to lint using git:')
+
+
+def test_get_yaml_files_silent(monkeypatch, capsys):
+    options, _ = cli.get_config([])
+    test_dir = os.path.dirname(os.path.realpath(__file__))
+    role_path = os.path.join(test_dir, 'roles/test-role')
+    monkeypatch.chdir(role_path)
+    files = utils.get_yaml_files(options)
+    stdout = capsys.readouterr().out
+    assert not stdout
+    assert len(files)
```

Comment:
The semantical meaning of these assertions is not quite obvious:
```suggestion
    assert not stdout, 'No output is expected when the verbosity is off'
    assert len(files), 'The fallback is expected to find some files'
```
(but was is fallback or git? this could be a great case for `@pytest.mark.parametrize` with the same asserts but with git binary present&git repo, no binary or no repo)